### ISSUE TITLE
"Open with OniVim2" 

### ISSUE THREAD

**Issue by iryan2** [onivim/oni2#1000]
When I attempt to open a file with OniVim2, the app launches but the file isn't opened. Instead I'm left with a blank "untitled" document. Is this currently expected to work? 

I'd love to be able to use Oni as my default editor for .txt and .svg files, even at this early stage. 
![open-with-onivim2](https://user-images.githubusercontent.com/5858312/70099762-85cfd480-167b-11ea-9291-4eb24f4a232c.gif)

**Comment by bryphe** [comment]
Thanks for opening the issue and the gif, @iryan2 ! Can confirm this bug - we should be able to open that way. 

It looks like we aren't handling the specific arguments from finder correctly in this case.

**Comment by yangziy** [comment]
It's been a long time but the bug is still around.

**Comment by KevinSjoberg** [comment]
@bryphe I'll look into this. 👍

**Comment by bryphe** [comment]
Awesome, thanks a lot for the help @KevinSjoberg !

Some ideas / things that might help:
- Here's the steps to create a 'release' `Onivim2.app` build: https://onivim.github.io/docs/for-developers/building#release 
- With the `esy create-release` command, you'll get a `_release/Onivim2.app` in your `oni2` folder - you can point the `Open With` dialog to that.

We'll likely need to add some logging to figure out why we aren't getting the commands we expect. It might help to log the arguments we get prior to parse (these are available via [`Sys.argv`](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Sys.html) - we could log them out around here: https://github.com/onivim/oni2/blob/0a17c40a96d78b28807f432838a196453d3b247a/src/bin_editor/Oni2_editor.re#L88)

The tricky thing about these startup-issues is that we don't get to add parameters to generate a log file - so we could temporarily hard-code our logging to always log to a file for the purpose of the investigation: https://github.com/onivim/oni2/blob/0a17c40a96d78b28807f432838a196453d3b247a/src/Core/Cli.re#L109

That way, when we start the app via 'Open With', we could still get a log we could work with.

Hope that gives some ideas! Thanks again and let me know if you have questions

**Comment by bryphe** [comment]
Oh, one other idea too - if we populate an `LSItemContentTypes` entry in our `Info.plist` - it'll help streamline the 'Open With ->' experience too:

https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html

We already specify a list of extensions + roles + types in our `package.json`: https://github.com/onivim/oni2/blob/0a17c40a96d78b28807f432838a196453d3b247a/package.json#L127

We could take that information, and then, in our `release.js` script (the script that handles `esy create-release`) - we could add it to our plist creation:
https://github.com/onivim/oni2/blob/0a17c40a96d78b28807f432838a196453d3b247a/scripts/release.js#L116

**Comment by KevinSjoberg** [comment]
Adding this for reference, https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html#//apple_ref/doc/uid/TP40009259-SW1.

**Comment by KevinSjoberg** [comment]
Unfortunately, I have not managed to get Oni2 to open the selected file yet, looking at the logs, we don't seem to get any command line arguments at all when opening a file using `Open With`.

Excerpt taken from the logs

```
[INFO]  Oni2_editor Starting Onivim 2.0.3.0 (cc79c6fe)
[INFO]  Oni2_editor Starting with arguments 1
[INFO]  Oni2_editor Argument: /Users/kevinsjoberg/code/oni2/_release/Onivim2.app/Contents/MacOS/Oni2_editor
```

Does Mac OS handle it differently?

**Comment by zymergen-luke** [comment]
Is there a linked ticket in Revery that we can watch?

**Comment by bryphe** [comment]
@zbaylin has actually just started working on this for us (integrating a hook in Revery for this API, and wiring it up into Onivim 2)

**Comment by KevinSjoberg** [comment]
@bryphe thanks for taking it over. I completely dropped the ball on this one. I recently became a father of two, so I've been busy with life. 🙂

Is there anything I can do here or is it pretty much done?

**Comment by bryphe** [comment]
@KevinSjoberg - no worries, my apologies - I think you were blocked and waiting on me. Thanks for exploring this and checking in.

> I recently became a father of two, so I've been busy with life. 🙂

Congratulations, that's so exciting! That's the good kind of busy 😄 

> Is there anything I can do here or is it pretty much done?

I think this is pretty much done, now - is there anything left @zbaylin , or can we close this out?

**Comment by zbaylin** [comment]
@KevinSjoberg Congratulations from me as well!

@bryphe I think we're done with this, it works well for me. If someone has a bug with the implement they can open a separate issue.

Glad to have this in!

**Comment by bryphe** [comment]
> If someone has a bug with the implement they can open a separate issue.

Sounds good, I'll close this out.
